libs/langchain/langchain/tools/gmail/get_message.py

```diff
@@ -33,6 +33,27 @@ def _run(
         run_manager: Optional[CallbackManagerForToolRun] = None,
     ) -> Dict:
         """Run the tool."""
+
+        def decode_payload(payload: bytes, charset: Optional[str] = None) -> str:
+            encodings_to_try = [
```

Comment:
Have you considered exposing these as parameters tot he tool? 

Here's a version I had  --  https://github.com/langchain-ai/langchain/compare/master...rescana-com/master

It could also allow swapping out the hard-coded list with something that infers the encoding